server/app.js

```diff
@@ -118,24 +120,46 @@ app.post("/webhook", (req, res) => {
       payloadToSign,
     });
 
+    // Validates if the timestamp is within an acceptable timeframe
+    // For more information visit https://webhooks.fyi/security/replay-prevention
+    const isTimestampValid = validateTimestamp(revolutRequestTimestamp);
+
+    if (!isTimestampValid) {
+      // The request timestamp is outside of the tolerance zone.
+      return res.status(403).send("Timestamp outside the tolerance zone");
+    }
+
     if (isSignatureValid) {
+      const order = orders.getOrderByRevolutId(req.body["order_id"]);
+
       // Return 200 response before doing any order computation
       res.sendStatus(200);
```

Comment:
now that there's an async queue, let's maybe schedule order handling and then return 200.
We can also add a comment above scheduling saying something like `Don't process orders directly in the webhook handler. Schedule order management logic (send emails, update orders, etc) and return 200 as soon as possible`

Reply:
we can even just push the event to the queue as is, and then process the order in the worker

Reply:
Agree, updated